src/agenteval/log.py

```diff
@@ -37,24 +38,52 @@ def collect_model_usage(sample: EvalSample) -> list[ModelUsageWithName]:
 def compute_model_cost(model_usages: list[ModelUsageWithName]) -> float:
     """
     Compute aggregate cost for a list of ModelUsageWithName objects.
-    Takes into account cached input tokens through cost_per_token's cache_read_input_tokens parameter.
+    Handles cached tokens via litellm Usage object.
     """
     total_cost = 0.0
     for model_usage in model_usages:
-
-        prompt_tokens = model_usage.usage.input_tokens
-        # anthropic results split total tokens across input, output, cache write, and cache read values
-        if model_usage.usage.total_tokens != prompt_tokens + model_usage.usage.output_tokens:
-            prompt_tokens += (model_usage.usage.input_tokens_cache_write + model_usage.usage.input_tokens_cache_read)
+        input_tokens = model_usage.usage.input_tokens
+        output_tokens = model_usage.usage.output_tokens
+        
+        cache_read_input_tokens = model_usage.usage.input_tokens_cache_read
+        # prompt_tokens_details = PromptTokensDetails(cached_tokens=cache_read_input_tokens)
+        
+        cache_write_input_tokens = model_usage.usage.input_tokens_cache_write
         
+
         try:
+            # input tokens count includes any cached tokens
+            if input_tokens == model_usage.usage.total_tokens - model_usage.usage.output_tokens:
+                text_tokens = input_tokens - cache_read_input_tokens
+                prompt_tokens = input_tokens
+
+            # (anthropic) input tokens count excludes cache read and cache write tokens
+            elif input_tokens == model_usage.usage.total_tokens - output_tokens - cache_read_input_tokens - cache_write_input_tokens:
+                text_tokens = input_tokens
+                prompt_tokens = input_tokens + cache_read_input_tokens + cache_write_input_tokens
```

Comment:
This is actually a bit different than what I understood from your earlier [comment](https://github.com/allenai/agent-eval/pull/11#issuecomment-2957368934):

> Hand calculation of anthropic cost for one litqa sample is significantly lower than the returned litellm cost.
> If I set prompt_tokens = input_tokens + cache_read_input_tokens, I get a result that matches the hand calculated result 

So should this be `prompt_tokens = input_tokens + cache_read_input_tokens`?  As long as the result matches hand calculation it's all good, just making sure I'm not getting my wires crossed.